astropy/utils/tests/test_collections.py

```diff
@@ -31,3 +33,35 @@ def test_homogeneous_list4():
 def test_homogeneous_list5():
     l = collections.HomogeneousList(int, [1, 2, 3])
     l[1] = 5.0
+
+
+def test_homogeneous_list_setitem_works():
+    l = collections.HomogeneousList(int, [1, 2, 3])
+    l[1] = 5
+
+
+def test_homogeneous_list_setitem_works_with_slice():
+    l = collections.HomogeneousList(int, [1, 2, 3])
+    l[0:1] = [10, 20, 30]
+
+
+def test_homogeneous_list_init_got_invalid_type():
+    with pytest.raises(TypeError):
+        collections.HomogeneousList(int, [1, 2., 3])
+
+
+def test_homogeneous_list_works_with_generators():
+    hl = collections.HomogeneousList(int, (i for i in range(3)))
```

Comment:
Does it not take generators as-is (without the list comprehension part)?

Reply:
It does, but I changed it to use `extend` internally, this was just to make sure it works.